src/main/java/nl/tudelft/controller/collision/CollisionHelper.java

```diff
@@ -29,7 +29,8 @@
             throw new IllegalArgumentException();
         } else {
             List<AbstractGameObject> returnObjects = new ArrayList<>();
-            quad.retrieve(returnObjects, objectA.getBounds());
+
+            returnObjects = quad.retrieve(returnObjects, objectA.getBounds());
```

Comment:
This seems all right yes. Passing the list that is being returned is pretty pointless anyway.
Returning it is more encapsulated and better practice.
Since the method is used recursively, simply removing the parameter and starting with an empty list will probably not work, but in my opinion this should be handled in the recursive logic (such as appending the returned list to the earlier obtained list).

But  Idon't think we should make such structural changes to an architecture that we don't fully understand.

< review